lib/utils/aws/aws.go

```diff
@@ -348,37 +344,87 @@ func isJSON(contentType string) bool {
 }
 
 // BuildRoleARN constructs a string AWS ARN from a username, region, and account ID.
-func BuildRoleARN(username, region, accountID string) string {
+func BuildRoleARN(username, region, accountID string) (string, error) {
```

Comment:
Could you add a bit more comment to this function and explain why the username values can be either full ARN only user name or `role` name ?